src/EditorFeatures/CSharp/Highlighting/KeywordHighlighters/DiscardParameterHighlighter.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editor.Implementation.Highlighting;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.Highlighting.KeywordHighlighters
+{
+    [ExportHighlighter(LanguageNames.CSharp)]
+    internal class DiscardParameterHighlighter : AbstractKeywordHighlighter
+    {
+        [ImportingConstructor]
+        public DiscardParameterHighlighter()
+        {
+        }
+
+        protected override bool IsHighlightableNode(SyntaxNode node)
+        {
+            if (!node.IsKind(CodeAnalysis.CSharp.SyntaxKind.Argument) || !(node is ArgumentSyntax syntax))
+            {
+                return false;
+            }
+
+            if (!syntax.RefOrOutKeyword.IsKind(SyntaxKind.OutKeyword))
+            {
+                return false;
+            }
+
+            if (syntax.Expression.IsKind(SyntaxKind.IdentifierName) && syntax.Expression is IdentifierNameSyntax name)
```

Comment:
initial kind check unnecessary.

Reply:
Shouldn't the kind check be faster for non matching cases since the short circuit prevents the type check from executing?

Reply:
Typechecks in keyword-highlighting are not a hotspot. It's fine to do a typechecks :-)

Reply:
that makes sense i guess. changed it in the latest commit.